### ISSUE TITLE
Moving Core Data

### ISSUE THREAD

**Issue by jhergott** [uchicago-mobi/2015-Spring-Forum#38]
So I made two projects called GameStore and JRHStore.  GameStore is where I handle the import, tokenizing and persistence for the Core Data.  JRHStore is going to be my "real" app.  Sorry for the confusing names.

I created a model in JRHStore to match my GameStore model.  I then used a text editor and copy and pasted the text in GameStore.sqlite and GameStore.sqlite-shm into JRHStore.sqlite and JRHStore.sqlite-shm.  I can now fetch the data I originally stored in GameStore in my JRHStore app so it seems like it worked.  But aren't those Core Data objects just stored in the simulator on my computer?  I'm confused how the Core Data is going to be loaded for grading.  Likewise if I were to send it to the AppStore.

**Comment by tabinks** [comment]
Yes, they are stored on the simulator file systems, but that is why you need to copy them out.  You should copy the ".sqlite" core data store and make sure to "copy" them, not just reference them.  Once you have done this, they will be included in the applications main bundle.  Any files in the main bundle will be packaged with the app.

It's the same things as adding an image file from your desktop to your iOS app so it is available in the bundle.  In this case the file is just an .sqlite file.

**Comment by jhergott** [comment]
So I drag and drop GameStore.sqlite, GameStore.sqlite-shm and GameStore.sqlite-wal into my JRHStore project?  Does it matter where I copy them to?

**Comment by tabinks** [comment]
Yes, just drag and drop.  Make sure they are located in your app bundle, the  exact location (i.e. a subfolder, etc.) is up to you.

**Comment by jhergott** [comment]
I'm confused what happens next... Do we not use Core Data in our "real" app?  Do we just use the .sqlite file?  Or is there an easy way to save that .sqlite file to Core Data?

**Comment by tabinks** [comment]
You need to use core data in both apps.  You build your data store in App 1 and copy the store to App 2.  In App 2, you need core data to read the store.  

The .sqlite file **is** the core data store.

**Comment by jhergott** [comment]
Added

```
if (!NSFileManager.defaultManager().fileExistsAtPath(url.path!)) {
            let defaultStorePath = NSBundle.mainBundle().pathForResource("GameStore", ofType:"sqlite")
            NSFileManager.defaultManager().copyItemAtPath(defaultStorePath!, toPath:url.path!, error: nil)
        }
```

in

```
lazy var persistentStoreCoordinator: NSPersistentStoreCoordinator?
```

and now I'm getting it to work.

Thank you!

**Comment by alextsu** [comment]
A follow-up question:

I was able to use the code in Jake's last comment within my Application Target, but when I created a new embedded framework today and copied the same code in, I found that 
```swift
<code>let defaultStorePath = NSBundle.mainBundle().pathForResource("GameStore", ofType:"sqlite")
```
Would always set defaultStorePath to nil.

I even double checked that my sqlite file was in the "Copy Bundle Resources" tab under targets for my embedded framework.

My question is, how do I get the "defaultStorePath" from my embedded framework?
